src/sources/soundsourceproxy.cpp

```diff
@@ -348,16 +348,26 @@ ExportTrackMetadataResult
 SoundSourceProxy::exportTrackMetadataBeforeSaving(Track* pTrack) {
     DEBUG_ASSERT(pTrack);
     const auto trackFile = pTrack->getFileInfo();
-    mixxx::MetadataSourcePointer pMetadataSource =
+    const mixxx::SoundSourcePointer pSoundSource =
             SoundSourceProxy(trackFile.toUrl()).m_pSoundSource;
-    if (pMetadataSource) {
-        return pTrack->exportMetadata(pMetadataSource);
-    } else {
+    if (!pSoundSource ||
```

Comment:
We need to apply the same algorithm as used in openAudioSource(). This function has to be generalized to work for this case by omitting to wrap the result into AudioSourceTrackProxy.

Reply:
I will prepare a solution that can be integrated as a prerequisite.

Reply:
Rebased on #3413.